### DISCUSSION TITLE
Retrigger Build Button

### DISCUSSION THREAD

**Discussion by matthewpereira** [concourse/concourse#6147] [Show and tell]
I was looking at the 'Retrigger Build' button the other day:

![Screen Shot 2020-10-09 at 11 31 1](https://user-images.githubusercontent.com/6372810/95606543-78684100-0a28-11eb-9028-03bef150e3ea.png)

It's just a bit different than the 'Add Build' button it sits next to. They were added at different times by different people, and UI elements drift over time in any project. The + is a bit soft at the edges and isn't the same size as the 'Add Build' button's +, and it's a bit wider than it is tall.

![Screen Shot 2020-10-09 at 11 45 34 AM](https://user-images.githubusercontent.com/6372810/95607242-7783df00-0a29-11eb-9234-cd33137114a5.png)

When we put it on a pixel grid we can see that the edges don't quite line up; I think it'd be cool to line up the vectors to sharpen up the lines, and make it a little more circular and symmetrical. I'd also like to see it at the same scale as the Add Build button that currently sits next to it in the UI.

![Screen Shot 2020-10-09 at 12 10 00 PM](https://user-images.githubusercontent.com/6372810/95606464-5cfd3600-0a28-11eb-92ce-0b5aef2b5b10.png)

I whipped up a replacement icon (see the 'Proposed Retrigger Build' option in the following screenshots) as well as an alternate that I'd love to get folks' feedback on. Does the current icon work better to signify the 'retrigger' concept, or do y'all like the single circular arrow alternate instead? Also, any thoughts on sharp triangle arrows, rounded triangle arrows, or line-triangle arrows?

![Vector Preview](https://user-images.githubusercontent.com/6372810/95607861-553e9100-0a2a-11eb-9c9e-2c366b3d2759.png)

Here's a preview of pixel rendering at 2x (a 'Retina' display):

![2x preview](https://user-images.githubusercontent.com/6372810/95607920-6d161500-0a2a-11eb-8fb9-fec2e8b0a898.png)

And here's how they render on a regular HD or lower display at 1x:

![1x preview](https://user-images.githubusercontent.com/6372810/95607924-70110580-0a2a-11eb-89f5-87fc581ed4fd.png)

Nice and crisp! ✨ 

There's plenty to still nit-pick over in this rough stage, but I don't want to go overboard before I get feedback about which direction to move in. 🙇

**Comment by agurney** [comment]
I like the alternate for being a bit closer to a "standard" reload/retry icon. No strong feelings on clockwise/anticlockwise at this point ;-)

This does make me think about recognizing what the circle-plus button does, for someone who's not already familiar. I know people have been confused by:
- the "play" button does not cause a build to happen
- if the pipeline is paused and you press circle-plus, something seems to happen but then gets stuck ("checking pipeline is not paused" sounds like it is still trying to figure out whether the pipeline is paused, not that it _definitely is_ and needs to be unpaused in order for the build to proceed)
- if you happen to be looking at an old build, then circle-plus and arrows-plus do very different things, and it may not be immediately clear what has happened
- people may not naturally make the connection between "I want stuff to _happen_" and "what I need to tell Concourse to do is add a new build to the list of builds"

If the rerun button did not have the plus at all, but was just the circular arrow, then it would look more distinct to circle-plus, and might read more naturally as a retry. I am not sure if it's a pro or con that it would match the "check now" circular arrow on a resource page; there isn't really a notion of rerunning an old check, and in some sense "check now" is closer to "make a new build" in that it is working with the latest version of everything.

To smash this can of worms fully open, I wonder whether circle-plus is better off being replaced as well, with something that is more suggestive of triggering an action right now. How about https://materialdesignicons.com/icon/airplane-takeoff ?

**Reply by matthewpereira** [reply]
Thanks for this list!

I agree that this is an area where getting the 'right' icon is bound to be tricky business - airplane metaphors are great fun, and add a lot of great character, but could also be confusing. I like the idea of just adding text to remove any ambiguity - just tell people what will happen if/when they click a button. Tooltips are a band-aid solution, but better than nothing - hopefully in the future we can rely on them to add additional context rather than relying on them solely to communicate function.

I think there's a fair bit to be improved by localizing the 'Trigger Build' button to the list of builds (in the position adjacent to the newest build) and localizing the 'Re-trigger Build' button to the selected build. I think there's an issue kicking around with some mockups to that effect - I'll try to track that down.
